service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java

```
@@ -60,4 +64,97 @@ public void changePassword(JWTData userData, ChangePasswordRequest changePasswor
       throw new WrongPasswordException();
     }
   }
+
+  @Override
+  public void setContactsAndChildren(JWTData userData, SetContactsAndChildrenRequest setContactsAndChildrenRequest) {
+
+    updateMainContact(userData, setContactsAndChildrenRequest.getMainContact());
+
+    if (setContactsAndChildrenRequest.getChildren() != null &&
+            !setContactsAndChildrenRequest.getChildren().isEmpty()) {
+      addChildren(setContactsAndChildrenRequest.getChildren(), userData);
+    }
+
+    if (setContactsAndChildrenRequest.getAdditionalContacts() != null &&
+            !setContactsAndChildrenRequest.getAdditionalContacts().isEmpty()) {
+      addAdditionalContacts(setContactsAndChildrenRequest.getAdditionalContacts(), userData);
+    }
+
+  }
+  
+  private void updateMainContact(JWTData userData, Contact newContactData) {
+
+    ContactsRecord mainContact = db.selectFrom(CONTACTS)
+            .where(CONTACTS.USER_ID.eq(userData.getUserId())
+                    .and(CONTACTS.IS_MAIN_CONTACT))
+            .fetchOne();
+
+    if (mainContact == null) {
+      return;
```

This seems like the wrong behavior.... I'd think we would either want to throw an exception or create a new record. Leaning towards throwing an exception since this shouldn't happen in our normal flow but open to opinion